src/db.cc

```diff
@@ -327,14 +327,39 @@ namespace Astroid {
     return _mid;
   }
 
-  ustring Db::add_sent_message (ustring fname, vector<ustring> additional_sent_tags) {
+  ustring Db::add_sent_message (ustring fname, vector<ustring> additional_sent_tags, ustring parent_mid) {
     LOG (info) << "db: adding sent message: " << fname;
     additional_sent_tags.insert (additional_sent_tags.end (), sent_tags.begin (), sent_tags.end ());
+
+    if (!parent_mid.empty () &&
+            find(additional_sent_tags.begin(), additional_sent_tags.end(), "*") != additional_sent_tags.end()) {
+        notmuch_message_t * parent_msg;
```

Comment:
Probably better to use `on_message` here, takes care of cleaning up the parent_msg. Might also be better to instantiate a `NotmuchMessage` on `parent_msg`, there is already a `get_tags` function which is more robust. `NotmuchMessage` should be cheap objects.

Reply:
Thanks, I'll have a look at that.